homeassistant/components/schedule/__init__.py

```diff
@@ -0,0 +1,310 @@
+
+    async def _process_create_data(self, data: dict) -> dict:
+        """Validate the config is valid."""
+        self.SCHEMA(data)
+        return data
+
+    @callback
+    def _get_suggested_id(self, info: dict) -> str:
+        """Suggest an ID based on the config."""
+        name: str = info[CONF_NAME]
+        return name
+
+    async def _update_data(self, data: dict, update_data: dict) -> dict:
+        """Return a new updated data object."""
+        self.SCHEMA(update_data)
+        return data | update_data
+
+    async def _async_load_data(self) -> dict | None:
+        """Load the data."""
+        if data := await super()._async_load_data():
+            data["items"] = [STORAGE_SCHEMA(item) for item in data["items"]]
+        return data
+
+
+class Schedule(Entity):
+    """Schedule entity."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+    _attr_state: Literal["active", "inactive"]
+    _config: ConfigType
+    _next: datetime
+    _unsub_update: Callable[[], None] | None = None
+
+    def __init__(self, config: ConfigType, editable: bool = True) -> None:
+        """Initialize a schedule."""
+        self._config = STORAGE_SCHEMA(config)
+        self._attr_capability_attributes = {ATTR_EDITABLE: editable}
+        self._attr_icon = self._config.get(CONF_ICON)
+        self._attr_name = self._config[CONF_NAME]
+        self._attr_unique_id = self._config[CONF_ID]
+
+    @classmethod
+    def from_yaml(cls, config: ConfigType) -> Schedule:
+        """Return entity instance initialized from yaml storage."""
+        schedule = cls(config, editable=False)
+        schedule.entity_id = f"{DOMAIN}.{config[CONF_ID]}"
+        return schedule
+
+    async def async_update_config(self, config: ConfigType) -> None:
+        """Handle when the config is updated."""
+        self._config = STORAGE_SCHEMA(config)
+        self._attr_icon = config.get(CONF_ICON)
+        self._attr_name = config[CONF_NAME]
+        self._clean_up_listener()
+        self._update()
+
+    @callback
+    def _clean_up_listener(self) -> None:
+        """Remove the update timer."""
+        if self._unsub_update is not None:
+            self._unsub_update()
+            self._unsub_update = None
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        self.async_on_remove(self._clean_up_listener)
+        self._update()
+
+    @callback
+    def _update(self, _: datetime | None = None) -> None:
+        """Update the states of the schedule."""
+        now = dt_util.now()
+        todays_schedule = self._config.get(WEEKDAY_TO_CONF[now.weekday()], [])
+
+        # Determine current schedule state
+        self._attr_state = next(
+            (
+                STATE_ACTIVE
+                for time_range in todays_schedule
+                if time_range[CONF_FROM] <= now.time() <= time_range[CONF_TO]
+            ),
+            STATE_INACTIVE,
```

Comment:
Should we use `idle` here? It is an easier word and more easier to distinguish from `active` when scanning a list.

Reply:
🤷 Idle sounds like a device state.

Reply:
As discussed on Discord, changing it to On/Off.